File: Data Structures & Algorithms/daily-temperatures/submission-0.cpp
class Solution {
public:
    vector<int> dailyTemperatures(vector<int>& temperatures) {
        stack<pair<int,int>>s;
        vector<int>ans;
        ans.push_back(0);
        s.push({temperatures[temperatures.size()-1],temperatures.size()-1});
        int j=temperatures.size()-2;
        while(j>=0){
            
                if(s.top().first>temperatures[j]){
                    ans.push_back(s.top().second-j);
                } else {
                    while(s.size()!=0 && s.top().first<=temperatures[j]){
                        s.pop();
                    }
                    if(s.size()!=0 && s.top().first>temperatures[j]){
                        ans.push_back(s.top().second-j);
                    }
                    else if(s.size()==0){
                        ans.push_back(0);
                    }
                }
                s.push({temperatures[j],j});
                j--;
            }
             reverse(ans.begin(),ans.end());
             return ans;
        }
    
};

File: Data Structures & Algorithms/pow-x-n/submission-0.java
class Solution {
    public double myPow(double x, int n) {
        int nn=n;
        if(n<0)nn=n*-1;
        double ans=1;
        while(nn>0){
            if(nn%2==1){
                ans*=x;
                nn-=1;
            } else {
                x*=x;
                nn/=2;
            }
        }
        if(n<0)return 1/(double)ans;
        return ans;
    }
}
